classes/publication/Repository.php

```diff
@@ -293,14 +295,18 @@ public function add(Publication $publication): int
      * Makes a copy of an existing publication, without the datePublished,
      * and makes copies of all associated objects.
      */
-    public function version(Publication $publication): int
+    public function version(Publication $publication, Submission $submission, Context $context, User $editor): int
     {
```

Comment:
I'm not using it, but @NateWr suggested it be added now since we were already changing the function signature as it would likely be helpful to have the associated user for hooks/events/etc.

Reply:
@asmecher has pushed back on this idea in our discussion about events here: https://github.com/pkp/pkp-lib/discussions/8388#discussioncomment-4040677

Maybe until we settle on an approach, it's better to load the dependencies you need inside the event listener. It won't be as performant but it is easier to add arguments to a function later than it is to remove them.

Reply:
This is also a place in the code where there is no event... So yes, I would remove it...